src/app/core/effects/now-playlist.effects.ts

```diff
@@ -36,11 +42,21 @@ export class NowPlaylistEffects {
   loadNextTrack$ = this.actions$
     .ofType(NowPlaylistActions.MEDIA_ENDED)
     .map(toPayload)
-    .withLatestFrom(this.store.let(getSelectedMedia$))
-    .filter((states: [any, GoogleApiYouTubeVideoResource]) => states[1] && states[1].hasOwnProperty('id'))
-    .map((states: [any, GoogleApiYouTubeVideoResource]) => {
-      return this.nowPlaylistActions.selectVideo(states[1]);
-    }).share();
+    .withLatestFrom(this.store.let(getNowPlaylist$))
+    .map(([action, nowPlaylist]: [any, NowPlaylistInterface]) => {
```

Comment:
what this function does? 
anyway, logics should be outside of this effect - if possible inside a reducer file (as an exported function at least).